unlock-app/src/hooks/useKeyGranter.ts

```diff
@@ -16,4 +18,60 @@ export const useKeyGranter = ({ network }: { network: number }) => {
   })
 }
 
+export const useAddKeyGranter = (lockAddress: string, network: number) => {
+  const { getWalletService } = useAuth()
+  const queryClient = useQueryClient()
+
+  const addKeyGranter = async (address: string) => {
+    const keyGranterAddress = minifyAddress(address)
+    const walletService = await getWalletService(network)
+
+    const addKeyGranterPromise = walletService.addKeyGranter({
+      lockAddress,
+      keyGranter: address,
+    })
+
+    await ToastHelper.promise(addKeyGranterPromise, {
+      loading: `Adding ${keyGranterAddress} as Key Granter.`,
+      success: `${keyGranterAddress} added as Key Granter.`,
+      error: ` Could not add ${keyGranterAddress} as Key Granter.`,
+    })
+  }
+
+  return useMutation({
+    mutationFn: addKeyGranter,
+    onSuccess: () =>
+      queryClient.invalidateQueries({
+        queryKey: ['fetchKeyGranters', lockAddress, network],
+      }),
+  })
+}
+
+export const useRemoveKeyGranter = (lockAddress: string, network: number) => {
+  const { getWalletService } = useAuth()
+  const queryClient = useQueryClient()
+
+  const removeKeyGranter = async (keyGranter: string) => {
+    const walletService = await getWalletService(network)
```

Comment:
Let's show a confirmation screen before deleting please.Also, if this is the locksmith key granter, please let's show the following in the message: "If you remove this key granter, credit card purchases for your contracts will be disabled". Otherwise, please show "Please confirm you want to remove this key granter".

Thanks!